### ISSUE TITLE
os-theme-advanced : Missing "Delete" Icon under System -> Snapshots

### ISSUE THREAD

**Issue by drinn** [opnsense/plugins#4207]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [X] The title contains the plugin to which this issue belongs

**Describe the bug**
The os-theme-advanced theme is missing an icon to delete a given snapshot.

**To Reproduce**
1. Go to System -> Snapshots
2. On the right under commands, there are icons for Activate, Edit, and Clone, but Delete is missing as it is in the default theme.

**Expected behavior**
A Delete button present like it is in the default theme.

**Screenshots**
os-theme-advanced:
![image](https://github.com/user-attachments/assets/a2e28c6e-1c0a-4535-b634-c70f4564d7c0)

default theme:
![image](https://github.com/user-attachments/assets/9f9b02d6-1117-435e-a3c9-64ae88abf2a1)

**Relevant log files**
N/A

**Additional context**
N/A

**Environment**
OPNsense 24.7.3 (amd64).

**Comment by youring** [comment]
Same situation in Haproxy list view, and the Add/Delete button seems narrower than usual.
![image](https://github.com/user-attachments/assets/13ff9bf5-dac9-4d5d-aafa-c38ed99ff48f)

**Comment by Jackysi** [comment]
Looking at the Settings -> Cron it seems table head has width of 7em. This causes issue with overflow that hiddes "Delete button". 
![image](https://github.com/user-attachments/assets/7e32d437-6f19-432b-a730-ef6704a1546b)

**Comment by Jackysi** [comment]
> [@Jackysi](https://github.com/Jackysi) are you still up for this? :)

Of course, but this isn't something I can fix on theme side? Since settings col width in HTML overrides CSS values there isn't much I can do. Unless I am missing something?

**Comment by fichtner** [comment]
@Jackysi hello! kind of, changing it in core will modify all even when not necessary, but:

@swhite2 I heard you were considering a sizing option for such cases as this?

**Comment by Jackysi** [comment]
> [@fichtner](https://github.com/fichtner) the option would involve user input, however, the "default" should always show all commands with proper spacing.
> 
> If the theming influences the sizing of the buttons in a different way than core, there's not much we can do here except make it more spacious in all cases, which is not preferable.
> 
> As far as I can see the snapshots' commands width is hardcoded to `9em`, which is fine in core and the theme should probably account for that

It would be preferable to switch such widths into CSS itself. So it can be overridden. E.g. using class action-buttons and set 7-9em width there. 

Hardcoding width into HTML where you have possibility to theme UI, seems odd.

**Comment by swhite2** [comment]
> It would be preferable to switch such widths into CSS itself. So it can be overridden. E.g. using a class "action-buttons" and set 7-9em width there.
> 
> Hardcoding width into HTML where you have possibility to theme UI, seems odd.

The table plugin defines such properties via `data-*` attributes as the use-case is on a per-column basis (managing the width internally). Theme issues have rarely popped up because of it.

Moreover, each page may maintain a different set of commands and therefore different widths, so the problem isn't exactly generic.

I think the column width is beyond the scope of a theme plugin. If the theme causes a column to overflow, while the core product doesn't, the styling issue is likely not the width of the column.
